libraries/AP_OpenDroneID/AP_OpenDroneID.cpp

```diff
@@ -0,0 +1,759 @@
+{
+    memcpy(_uas_id,uas_id,20);
+}
+
+void AP_OpenDroneID::set_height_reference(MAV_ODID_HEIGHT_REF reference)
+{
+    //check if reference is valid
+    if ((reference == MAV_ODID_HEIGHT_REF_OVER_TAKEOFF) || (reference == MAV_ODID_HEIGHT_REF_OVER_GROUND)) {
+        _height_reference = reference;
+    }
+    else {
+        _height_reference = MAV_ODID_HEIGHT_REF_OVER_TAKEOFF;   //default reference
+    }
+}
+
+void AP_OpenDroneID::set_uav_status(MAV_ODID_STATUS status)
+{
+    //check if status is valid
+    if ((status == MAV_ODID_STATUS_GROUND) || (status == MAV_ODID_STATUS_AIRBORNE)
+        || (status == MAV_ODID_STATUS_EMERGENCY)) {
+        _uav_status = status;
+    }
+    else {
+        _uav_status = MAV_ODID_STATUS_UNDECLARED;
+    }
+}
+
+void AP_OpenDroneID::set_operator_location_type(MAV_ODID_OPERATOR_LOCATION_TYPE operator_location_type)
+{
+    //check if operator_location_type is valid
+    if ((operator_location_type == MAV_ODID_OPERATOR_LOCATION_TYPE_TAKEOFF) || (operator_location_type == MAV_ODID_OPERATOR_LOCATION_TYPE_LIVE_GNSS)
+        || (operator_location_type == MAV_ODID_OPERATOR_LOCATION_TYPE_FIXED)) {
+        _operator_location_type = operator_location_type;
+    }
+    else {
+        _operator_location_type = MAV_ODID_OPERATOR_LOCATION_TYPE_TAKEOFF;
+    }
+}
+
+void AP_OpenDroneID::set_classification_type(MAV_ODID_CLASSIFICATION_TYPE classification_type)
+{
+    //check if classification_type is valid
+    if (classification_type == MAV_ODID_CLASSIFICATION_TYPE_EU) {
+        _classification_type = classification_type;
+    }
+    else {
+        _classification_type = MAV_ODID_CLASSIFICATION_TYPE_UNDECLARED;
+    }
+}
+
+void AP_OpenDroneID::set_category_eu(MAV_ODID_CATEGORY_EU category_eu)
+{
+    //check if category_eu is valid
+    if ((category_eu == MAV_ODID_CATEGORY_EU_OPEN) || (category_eu == MAV_ODID_CATEGORY_EU_SPECIFIC)
+        || (category_eu == MAV_ODID_CATEGORY_EU_CERTIFIED)
+    ) {
+        _category_eu = category_eu;
+    }
+    else {
+        _category_eu = MAV_ODID_CATEGORY_EU_UNDECLARED;
+    }
+}
+
+void AP_OpenDroneID::set_class_eu(MAV_ODID_CLASS_EU class_eu)
+{
+    //check if class_eu is valid
+    if ((class_eu == MAV_ODID_CLASS_EU_CLASS_0) || (class_eu == MAV_ODID_CLASS_EU_CLASS_1) || (class_eu == MAV_ODID_CLASS_EU_CLASS_2)
+        || (class_eu == MAV_ODID_CLASS_EU_CLASS_3) || (class_eu == MAV_ODID_CLASS_EU_CLASS_4) || (class_eu == MAV_ODID_CLASS_EU_CLASS_5)
+        || (class_eu == MAV_ODID_CLASS_EU_CLASS_6) ) {
+        _class_eu = class_eu;
+    }
+    else {
+        _class_eu = MAV_ODID_CLASS_EU_UNDECLARED;
+    }
+}
+void AP_OpenDroneID::set_operator_id_type(MAV_ODID_OPERATOR_ID_TYPE type)
+{
+    //check if type is valid
+    if (type == MAV_ODID_OPERATOR_ID_TYPE_CAA) {
+        _operator_id_type = type;
+    }
+    else {
+        _operator_id_type = MAV_ODID_OPERATOR_ID_TYPE_CAA;
+    }
+}
+
+void AP_OpenDroneID::set_operator_id(char* id)
+{
+    memcpy(_operator_id,id,20);
```

Comment:
This will set junk on strings length less than 20
```suggestion
    strncpy(_operator_id, id, sizeof(_operator_id));
```

And later when you populate the field, make sure to use strncpy() to fill the temp char[] when making the mavlink packet